lib/models/BucketInfo.js

```diff
@@ -387,6 +395,23 @@ class BucketInfo {
         this._objectLockConfiguration = objectLockConfiguration;
         return this;
     }
+    /**
+     * Get notification configuration
+     * @return {object|null} notification configuration information or 'null' if
+     * the bucket does not have a notification configuration
+     */
+    getNotificationConfiguration() {
+        return this._notificationConfiguration;
+    }
+    /**
+     * Set notification configuraiton
+     * @param {object} notificationConfiguration - bucket notification information
+     * @return {BucketInfo} - bucket info instance
+     */
+    setNotificationConfiguration(notificationConfiguration) {
+        this._notificationConfiguration = notificationConfiguration;
```

Comment:
When using the constructor the notification configuration is validated, but when setting it is not. Not sure if this discrepency matters but I think they should both be validated.

Reply:
in cloudserver, the put bucket notification api uses the constructor to validate a config before setting it

Reply:
I am sorry I do not understand. The put bucket notification uses the constructor to validate a config and then sets the config again after it was validated and stored already in the constructor?